src/webapp/components/replicate/ReplicateUserFromTemplate.tsx

```diff
@@ -0,0 +1,368 @@
+import React, { useCallback, useEffect } from "react";
+import _ from "lodash";
+
+import { useLoading, useSnackbar } from "@eyeseetea/d2-ui-components";
+import { Dialog, DialogTitle, DialogActions, DialogContent, Button } from "@material-ui/core";
+import {
+    InputField,
+    composeValidators,
+    createMaxCharacterLength,
+    createMinCharacterLength,
+    hasValue,
+    string,
+    number,
+} from "@dhis2/ui";
+import { Form, FormSpy, Field } from "react-final-form";
+import { FormState } from "final-form";
+
+import i18n from "../../../locales";
+import { useAppContext } from "../../contexts/app-context";
+import { getFromTemplate } from "../../../utils/template";
+
+import { Id } from "../../../domain/entities/Ref";
+import { User, defaultUser } from "../../../domain/entities/User";
+import { UserLogic } from "../../../domain/entities/UserLogic";
+
+interface ReplicateUserFromTemplateProps {
+    userToReplicateId: Id;
+    onRequestClose: () => void;
+}
+
+export const ReplicateUserFromTemplate: React.FC<ReplicateUserFromTemplateProps> = props => {
```

Comment:
The rest of the presentation logic (manage state, actions, invoke use cases, etc.) should be in a custom hook: useReplicateUserFromTemplate.

In this custom hook, we should use the user entity, not the user props